cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (C) 2014 Harm Weites
+# Copyright (C) 2019 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _select_hostname(self, hostname, fqdn):
+        if fqdn:
+            return fqdn
+        return hostname
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(filename='/etc/rc.conf')
+        return ('/etc/rc.conf', sys_hostname)
+
+    def _read_hostname(self, filename, default=None):
+        return bsd_util.get_rc_config_value('hostname')
+
+    def _write_hostname(self, hostname, filename):
+        bsd_util.set_rc_config_value('hostname', hostname,
```

Comment:
in freebsd, we use rhel_util, how is this different?

Reply:
on NetBSD, the `/etc/rc.conf` comes with a `if` block. The rhel_util parser cannot parse the file properly. And so I decided to write these two light setter/getter functions.

Reply:
Shallow person question: Is there enough alignment between netbsd and freebsd that one could a be a subclass of the other distro? (like ubuntu.py is a subclass of debian distro as it is a derivative work that can share the same distro class methods). As you've implemented this netbsd is a subclass only of top-level Distro class and thus has to redefine all methods.